backend/controllers/UserController.ts

```diff
@@ -31,34 +31,46 @@ const UserController = {
 
   async postUserLocation(req: Request, res: Response) {
     try {
-      const { locationName, description, coordinates, photoSrc } = req.body;
-      const _id = req.user;
-      const userData = await User.findById(_id);
+      const { locationName, description, coordinates, author } = req.body;
 
-      if (!userData) {
-        return res.status(400).json({ error: "User doesn't exist" });
-      }
+      const location = await Location.find({ coordinates: coordinates });
+
+      const imageUrls: string[] = [];
+
+      if (location.length === 0) {
+        Array.prototype.forEach.call(req.files, file => {
+          imageUrls.push(file.location);
+        });
 
-      const userLocation = new Location({
-        locationName: locationName,
-        coordinates: coordinates,
-        photoSrc: photoSrc,
-        description: description,
-        comments: [],
-        rating: {
-          likes: [],
-          dislikes: []
-        },
-        filters: {
-          favorite: [],
-          visited: [],
-          author: _id,
-          cost: 'free'
+        console.log(imageUrls, imageUrls);
+
+        const _id = req.user;
+        const userData = await User.findById(_id);
+
+        if (!userData) {
+          return res.status(400).json({ error: "User doesn't exist" });
         }
-      });
 
-      const result = await userLocation.save();
-      return res.status(200).json({ result });
+        const userLocation = new Location({
+          locationName: locationName,
+          coordinates: coordinates,
+          arrayPhotos: imageUrls,
+          description: description,
+          comments: [],
+          rating: {
+            likes: [],
+            dislikes: []
+          },
+          filters: [],
+          author: author
+        });
+
+        const result = await userLocation.save();
+
+        return res.status(200).json(result);
+      } else {
+        res.status(400).json({ error: 'Data is present' });
+      }
     } catch (err: any) {
       return res.status(500).json({ error: err.message });
     }
```

Comment:
this method should be replaced in location controller

Reply:
Done